### PR TITLE
Remove last uses of old Azure SDK for Go

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Removes the final two vestiges of the "track 1" version of Azure SDK for Go in the CAPZ codebase.

**Which issue(s) this PR fixes**:
Refs #3409

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -58,7 +56,7 @@ func Get() Info {
 		GoVersion:       runtime.Version(),
 		Compiler:        runtime.Compiler,
 		Platform:        fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
-		AzureSdkVersion: azuresdkversion.Number,
+		AzureSdkVersion: "SDK track2",
```
Comment (mboersma) [inline version/version.go:59]: There is no single version string provided any longer for the SDK (except arguably in an `internal` package that's not accessible). So I think all we can do is hard-code it from now on.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**